Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,236 @@
+
+                string currentBalanceHolder = userHistory[0];
+
+                for (int i = 1; i < userHistory.Length - 1; i++)
+                {
+                    balance = double.TryParse(userHistory[i], out double result) ?
+                        double.Parse(userHistory[i]) : double.NaN;
+
+                    nextBalance = double.TryParse(userHistory[i + 1], out double nextResult) ?
+                        double.Parse(userHistory[i + 1]) : double.NaN;
+
+                    if ((nextBalance - balance) < biggestLoss)
+                    {
+                        biggestLoser = currentBalanceHolder;
+                        holderCount = 1;
+                    }
+
+                    if (biggestLoss == nextBalance - balance)
+                    {
+                        if ( holderCount > 1)
+                        {
+                            biggestLoser += ($" and {currentBalanceHolder}");
+                        }
+                        if (holderCount == 1)
+                        {
+                            biggestLoser += ($", {currentBalanceHolder}");
+                            holderCount++;
+                        }
+                    }
+                    biggestLoss = ((nextBalance - balance) < biggestLoss) ?
+                        (nextBalance - balance) : biggestLoss;
+                }
+            }
+            string stringBiggestLoss = Math.Abs(biggestLoss).ToString();
+            return Invariant($"{biggestLoser} lost the most money. -¤{stringBiggestLoss}.");
+        }
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+            int holderCount = 0;
+            double highestLastBalance = double.MinValue;
+            string currentBalanceHolder;
+            string highestLastBalanceHolder = "";
+            foreach (string entry in peopleAndBalances)
+            {
+                string[] userHistory = entry.Split(',');
+
+                currentBalanceHolder = userHistory[0];
+
+                double currentBalanceHolderFinalBalance = double.Parse(userHistory[^1]);
+
+                if (highestLastBalance < currentBalanceHolderFinalBalance)
+                {
+                    highestLastBalanceHolder = currentBalanceHolder;
+                    holderCount = 1;
+                }
+
+                if (highestLastBalance == currentBalanceHolderFinalBalance &&
+                    highestLastBalanceHolder != currentBalanceHolder &&
+                    holderCount == 2)
+                {
+                    highestLastBalanceHolder += ($" and {currentBalanceHolder}");
+                }
+                if (highestLastBalance == currentBalanceHolderFinalBalance &&
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Do not check floating point equality with exact values, use a range instead.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4953253)